tests/test_server.py

```diff
@@ -634,6 +635,52 @@ async def test_train_with_yaml(rasa_app: SanicASGITestClient, tmp_path: Path):
     assert_trained_model(response.body, tmp_path)
 
 
+@pytest.mark.parametrize(
+    "params", [{}, {"augmentation": 20, "num_threads": 2, "force_training": True}]
+)
+async def test_train_with_yaml_with_params(
+    monkeypatch: MonkeyPatch,
+    rasa_app: SanicASGITestClient,
+    tmpdir: pathlib.Path,
+    params: Dict,
+):
+    fake_model = Path(tmpdir) / "fake_model.tar.gz"
+    fake_model.touch()
+    fake_model_path = str(fake_model)
+    future = asyncio.Future()
+    future.set_result(TrainingResult(model=fake_model_path))
+    mock_train = Mock(return_value=future)
+    monkeypatch.setattr(sys.modules["rasa.train"], "train_async", mock_train)
```

Comment:
we recently moved this:
```suggestion
    monkeypatch.setattr(rasa.model_training, "train_async", mock_train)
```